fairlearn/metrics/_plotter.py

```diff
@@ -0,0 +1,272 @@
+_CONF_INTERVALS_MUST_BE_TUPLE = "Calculated conf_intervals must be tuple of length 2"
+_ERROR_BARS_NEGATIVE_VALUE_ERROR = "Calculated relative errors from error_bars must be positive"
+_CONF_INTERVALS_FLIPPED_BOUNDS_ERROR = \
+    "Calculated conf_intervals' upper bound cannot be less than lower bound"
+
+
+def _check_if_metrics_and_error_metrics_same_length(metrics, error_bars, conf_intervals):
+    if error_bars is not None:
+        if len(error_bars) != len(metrics):
+            raise ValueError(_METRIC_AND_ERROR_BARS_NOT_SAME_LENGTH_ERROR)
+    elif conf_intervals is not None:
+        if len(conf_intervals) != len(metrics):
+            raise ValueError(_METRIC_AND_CONF_INTERVALS_NOT_SAME_LENGTH_ERROR)
+
+
+def _check_if_metrics_length_zero(metrics):
+    if len(metrics) == 0:
+        raise ValueError(_METRIC_LENGTH_ZERO_ERROR)
+
+
+def _check_for_ambiguous_error_metric(error_bars, conf_intervals):
+    # ambiguous error metric if both are provided
+    if error_bars is not None and conf_intervals is not None:
+        raise ValueError(_GIVEN_BOTH_ERROR_BARS_AND_CONF_INT_ERROR)
+
+
+def _check_for_valid_metrics_format(metrics):
+    # ensure metrics is either list, str, or None
+    if not (isinstance(metrics, list) or isinstance(metrics, str) or metrics is None):
+        raise ValueError(_METRICS_NOT_LIST_OR_STR_ERROR.format(type(metrics)))
+
+
+def _check_valid_error_bars(df_error_bars):
+    for tup in df_error_bars:
+        if not isinstance(tup, tuple) or len(tup) != 2:
+            raise ValueError(_ERROR_BARS_MUST_BE_TUPLE)
+        if not all(ele >= 0 for ele in tup):
+            raise ValueError(_ERROR_BARS_NEGATIVE_VALUE_ERROR)
+
+
+def _check_valid_conf_interval(df_conf_intervals):
+    for tup in df_conf_intervals:
+        if not isinstance(tup, tuple) or len(tup) != 2:
+            raise ValueError(_CONF_INTERVALS_MUST_BE_TUPLE)
+        if tup[0] > tup[1]:
+            raise ValueError(_CONF_INTERVALS_FLIPPED_BOUNDS_ERROR)
+
+
+def _plot_df(df, metrics, kind, subplots, df_all_errors=None, **kwargs):
+    if df_all_errors is not None:
+        yerr = np.array(
+            [np.array([[*row] for row in df_all_errors[metric]]).T for metric in metrics]
+        )
+        if kind == "scatter":
+            axs = df[metrics].plot(linestyle='', marker='o', yerr=yerr,
+                                   subplots=subplots, **kwargs)
+        else:
+            axs = df[metrics].plot(kind=kind, yerr=yerr, subplots=subplots, **kwargs)
+
+        if isinstance(axs, np.ndarray):
+            for ax in axs.flatten():
+                if kind == "scatter":
+                    color = ax.lines[0].get_color()
+                else:
+                    color = "black"
+
+                # extend legend with 95% CI text
+                handles, labels = ax.get_legend_handles_labels()
+                custom_line = [Line2D([0], [0], color=color, label="95% CI")]
```

Comment:
We should not be sayng `95% CI`. Maybe there should be an optional argument that the user can provide.

Reply:
I agree that we should not hardcode that it's a 95% CI, but I do feel that adding a legend item by default is appropriate; it might say just "CI" or "Confidence Interval". WDYT? I like the idea of the optimal argument.